lib/dependencies/ImportMetaPlugin.js

```diff
@@ -56,26 +56,27 @@ class ImportMetaPlugin {
 							"ImportMetaPlugin",
 							toConstantDependency(parser, JSON.stringify("object"))
 						);
-					parser.hooks.metaProperty.tap(
-						"ImportMetaPlugin",
-						toConstantDependency(parser, "Object()")
-					);
-					parser.hooks.metaProperty.tap("ImportMetaPlugin", metaProperty => {
-						const CriticalDependencyWarning = getCriticalDependencyWarning();
-						parser.state.module.addWarning(
-							new ModuleDependencyWarning(
-								parser.state.module,
-								new CriticalDependencyWarning(
-									"Accessing import.meta directly is unsupported (only property access is supported)"
-								),
-								metaProperty.loc
-							)
-						);
-						const dep = new ConstDependency("Object()", metaProperty.range);
-						dep.loc = metaProperty.loc;
-						parser.state.module.addPresentationalDependency(dep);
-						return true;
-					});
+					parser.hooks.metaProperty
```

Comment:
Do we even need a separate hooks for that or could we just use the `expression` hook?

Reply:
When I was doing this I also thought about it, I think it is about 2 things here:
1. perf - algorithm complexity is the same.
2. DX - I think for DX it is better to have separate hook since `MetaProperty` !== `Expression`